File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.java
class Solution {
    public String reverseParentheses(String s) {
		Stack<Character> stack = new Stack();
		Queue<Character> queue = new LinkedList();
		
		for(int i = 0; i < s.length(); i++) {
			if(s.charAt(i) == ')') {
				while(stack.peek() != '(') {
					queue.add(stack.pop());
				}
				stack.pop();
				while(queue.size() > 0) {
					stack.push(queue.remove());
				}
			} else {
				stack.push(s.charAt(i));
			}
		}
		
		char ans[] = new char[stack.size()];
		int indx = stack.size() - 1;
		
		while (stack.size() > 0) {
			ans[indx] = stack.pop();
			indx--;
		}
		
		String word = new String(ans);
		return word;		
	}
}

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        stack = []
        for i in s:
            if i != ')':
                stack.append(i)
            else:
                temp = []
                while stack[-1] != '(':
                    temp.append(stack.pop())
                stack.pop()
                while temp:
                    stack.append(temp.pop(0))
        return "".join(stack)